### ISSUE TITLE
MAUI WinUI Drag/Drop Not Working On Same UIElement

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/maui#13770]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/MAUI-WinUI-DragDrop-Not-Working-On-Same/10297499)._

---
I have ported some Xamarin UWP custom renderer code to MAUI handlers which is related to supporting Drag/Drop feature on AbsoluteLayout control which is mainly for reordering children inside AbsoluteLayout, but I have noticed that only DragStarting and DropCompleted events are getting fired but the rest of the Drag/Drop events are not fired on WinUI platform like DragEnter, DragOver, DragLeave and Drop events.

Same code used to work on Xamarin platform but its not working on WinUI.

In my code I try to drag/reorder the AbsoluteLayout children by dragging them and dropping them in different position inside same control (not to another control).
When I start the drag I create my own custom drag image and I have noticed that it shows operation not allowed icon while moving the item I am dragging around, so maybe that's a hint that could help.

I am attaching the demo code with this ticket (MauiDragDropApp.zip).

Steps to reproduce:
1- Run the app on Windows platform and you should see 3 rectangles.
2- Press on the first rectangle and start dragging it and you should see that non of the drag/drop events I listed above are getting fired.
3- You can check the visual studio output window to see the debug messages I wrote inside each of these events.

---
#### Feedback Bot on 3/2/2023, 07:06 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by mattleibow** [comment]
@TamerIrdais is this your issue? Does it work on other platforms?

**Comment by TamerIrdais** [comment]
> @TamerIrdais is this your issue? Does it work on other platforms?

Yes @mattleibow this is my issue and it works on other platforms, but you know each platform has its own implementation inside its handler code but I tried it on Android and it works there.

**Comment by TamerIrdais** [comment]
@mattleibow any update on this issue please?

**Comment by sam-wheat** [comment]
I am hitting this issue also and it is a drop dead blocker.  Any suggestions for a workaround or ETA for a fix?

Thanks

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 2.0. Can repro on windows platform with sample project.
[MauiDragDropApp.zip](https://github.com/dotnet/maui/files/11859017/MauiDragDropApp.zip)
![13770](https://github.com/dotnet/maui/assets/77434865/1a009f80-cfcf-493b-8001-8cf5a7b4a9b5)

**Comment by TamerIrdais** [comment]
Any update on this issue? any change of status or any comment from Microsoft?
It has been a year with no as much as a feedback or comment from Microsoft, @mattleibow?????

**Comment by TamerIrdais** [comment]
I can confirm that this issue is now fixed even on .NET 8.
